spec/cyclonedx/cocoapods/bom_builder_spec.rb

```diff
@@ -348,25 +363,49 @@
           components_generated_from_bom_builder = xml.at('bom/components')
 
           components = Nokogiri::XML(Nokogiri::XML::Builder.new(encoding: 'UTF-8') do |xml|
-            xml.components('xmlns': described_class::NAMESPACE) do
+            xml.components(xmlns: described_class::NAMESPACE) do
               bom_builder.pods.each { |pod| pod.add_to_bom(xml) }
             end
           end.to_xml).at('components')
 
           expect(components_generated_from_bom_builder).to be_equivalent_to(components)
         end
+
+        it 'should generate a child dependencies node' do
+          expect(xml.at('bom/dependencies')).not_to be_nil
+        end
+
+        it 'shoudl properly set dependencies node' do
+          dependencies_generated_from_bom_builder = xml.at('bom/dependencies')
+
+          dependencies = Nokogiri::XML dependencies_result
+
+          expect(dependencies_generated_from_bom_builder.to_xml).to be_equivalent_to(dependencies.root.to_xml)
+        end
       end
     end
 
     context 'without a component' do
       let(:bom_builder) { described_class.new(pods: pods) }
+      let(:dependencies_result) { '<dependencies/>' }
 
       it_behaves_like "bom_generator"
     end
 
     context 'with a component' do
       let(:component) { CycloneDX::CocoaPods::Component.new(name: 'Application', version: '1.3.5', type: 'application') }
-      let(:bom_builder) { described_class.new(component: component, pods: pods) }
+      let(:bom_builder) { described_class.new(component: component, pods: pods, dependencies: dependencies) }
+      let(:dependencies_result) do
+        '<dependencies>
+                                    <dependency ref="pkg:cocoapods/Alamofire@5.6.2"/>
+                                    <dependency ref="pkg:cocoapods/MSAL@1.2.1">
+                                      <dependency ref="pkg:cocoapods/MSAL@1.2.1#app-lib"/>
+                                    </dependency>
+                                    <dependency ref="pkg:cocoapods/FirebaseAnalytics@7.10.0#app-lib"/>
```

Comment:
Should these three purl expressions (lines 404-406) have `#app-lib` at the end of them?  It looks like that is a proper dependency purl for `MSAL/app-lib` but not the others.  The same question holds for new lines 272-274 above.  Looks like maybe something is being used in a loop, and being set once by the `MSAL/app-lib` pod and then being used incorrectly on the next three dependencies.

I don't see right now where that's happening, but I will try to take some time and determine why these show up.  Could be a problem in existing code, or might be a problem with the new dependency analysis code.

Reply:
@macblazer  fixed the problem, the issue was the dependencies declaration on the spec.